protobuf/lib/src/protobuf/pb_list.dart

```diff
@@ -6,310 +6,216 @@ part of protobuf;
 
 typedef CheckFunc<E> = void Function(E? x);
 
-class FrozenPbList<E> extends PbListBase<E> {
-  FrozenPbList._(List<E> wrappedList) : super._(wrappedList);
-
-  factory FrozenPbList.from(PbList<E> other) =>
-      FrozenPbList._(other._wrappedList);
-
-  UnsupportedError _unsupported(String method) =>
-      UnsupportedError('Cannot call $method on an unmodifiable list');
-
-  @override
-  void operator []=(int index, E value) => throw _unsupported('set');
-  @override
-  set length(int newLength) => throw _unsupported('set length');
-  @override
-  void setAll(int index, Iterable<E> iterable) => throw _unsupported('setAll');
-  @override
-  void add(E? element) => throw _unsupported('add');
-  @override
-  void addAll(Iterable<E> iterable) => throw _unsupported('addAll');
-  @override
-  void insert(int index, E element) => throw _unsupported('insert');
-  @override
-  void insertAll(int index, Iterable<E> iterable) =>
-      throw _unsupported('insertAll');
-  @override
-  bool remove(Object? element) => throw _unsupported('remove');
-  @override
-  void removeWhere(bool Function(E element) test) =>
-      throw _unsupported('removeWhere');
-  @override
-  void retainWhere(bool Function(E element) test) =>
-      throw _unsupported('retainWhere');
-  @override
-  void sort([Comparator<E>? compare]) => throw _unsupported('sort');
-  @override
-  void shuffle([math.Random? random]) => throw _unsupported('shuffle');
-  @override
-  void clear() => throw _unsupported('clear');
-  @override
-  E removeAt(int index) => throw _unsupported('removeAt');
-  @override
-  E removeLast() => throw _unsupported('removeLast');
-  @override
-  void setRange(int start, int end, Iterable<E> iterable,
-          [int skipCount = 0]) =>
-      throw _unsupported('setRange');
-  @override
-  void removeRange(int start, int end) => throw _unsupported('removeRange');
-  @override
-  void replaceRange(int start, int end, Iterable<E> newContents) =>
-      throw _unsupported('replaceRange');
-  @override
-  void fillRange(int start, int end, [E? fill]) =>
-      throw _unsupported('fillRange');
-}
+class PbList<E> extends ListBase<E> {
+  final List<E> _wrappedList;
+  final CheckFunc<E> _check;
 
-class PbList<E> extends PbListBase<E> {
-  PbList({CheckFunc<E> check = _checkNotNull}) : super._noList(check: check);
+  // TODO: Just store if the element type is a group or message?
```

Comment:
Always leave a name on the todos:
```suggestion
  // TODO(omersa): Just store if the element type is a group or message?
```

Reply:
I think storing the full type should be fine. In the lifetime of a `PbList` we only read this field once so it doesn't matter too much. Removed the TODO.